File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
class Solution {
    static String[] vowels = {"A", "E", "I", "O", "U"};
    static int count = 0;
    static int answer = 0;

    public int solution(String word) {
        dfs("", word);
        return answer;
    }

    private void dfs(String current, String word) {
        if (current.equals(word)) {
            answer = count;
            return;
        }

        if (current.length() == 5) return;

        for (int i = 0; i < 5; i++) {
            count++;
            dfs(current + vowels[i], word);
            if (answer != 0) return;
        }
    }
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.js
function solution(word) {
    const vowels = ['A', 'E', 'I', 'O', 'U'];
    let count = 0;
    let answer = 0;

    function dfs(current) {
        if (current === word) {
            answer = count;
            return;
        }

        if (current.length === 5) return;

        for (let i = 0; i < vowels.length; i++) {
            count++;
            dfs(current + vowels[i]);
            if (answer !== 0) return;
        }
    }

    dfs("");
    return answer;
}